cumulusci/tasks/push/tests/test_push_tasks.py

```diff
@@ -0,0 +1,481 @@
+            "version": VERSION,
+            "start_time": datetime.datetime.now(),
+        },
+    )
+    task._init_task()
+    task._init_options(
+        {"orgs": ORG_FILE, "version": VERSION, "start_time": datetime.datetime.now()}
+    )
+    assert task.options["namespace"] is None
+    assert task.options["batch_size"] == 200
+    assert task.options["orgs"] == ORG_FILE
+    assert task.options["version"] == VERSION
+    os.remove(ORG_FILE)
+
+
+def test_schedule_push_org_list_bad_start_time():
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "start_time": "2020-06-10T10:15",
+            "namespace": NAMESPACE,
+        },
+    )
+    task.push = mock.MagicMock()
+    with pytest.raises(CumulusCIException):
+        task._run_task()
+    os.remove(ORG_FILE)
+
+
+def test_load_orgs_file():
+    # creating sample org file for testing
+    # testing with empty file
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+        task = create_task(BaseSalesforcePushTask, options={})
+        assert task._load_orgs_file(ORG_FILE) == []
+
+    # testing with multiple orgs
+    with open(ORG_FILE, "r") as file:
+        assert task._load_orgs_file(ORG_FILE) == [
+            "00DS0000003TJJ6MAO",
+            "00DS0000003TJJ6MAL",
+        ]
+    os.remove(ORG_FILE)
+
+
+def test_get_subs_raises_err():
+    task = create_task(GetSubscriberList, options={"filename": "in.csv"})
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_filter_subs_raises_err():
+    task = create_task(
+        FilterSubscriberList, options={"file_in": "in.csv", "file_out": "out.txt"}
+    )
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_base_push_task_raises_err():
+    task = create_task(BaseSalesforcePushTask, options={})
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_report_push_status_error():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.sf = mock.MagicMock()
+    task.push_report = mock.MagicMock()
+    task.sf.query_all.return_value = {"totalSize": 1, "records": []}
+    with pytest.raises(PushApiObjectNotFound):
+        task._report_push_status("0DV1R000000k9dEWAQ")
+
+
+def test_get_push_request_job_results(
+    package_push_job_success, package_push_job_failure, package_push_job_cancel
+):
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.sf = mock.MagicMock()
+    task.push_report = mock.MagicMock()
+    task.push_request = mock.MagicMock()
+    task.push_request.get_push_job_objs.return_value = [
+        package_push_job_success,
+        package_push_job_failure,
+        package_push_job_cancel,
+    ]
+    task._get_push_request_job_results()
+    task.push_request.get_push_job_objs.assert_called_once()
```

Comment:
This would be a better test if it actually asserted that _get_push_request_job_results logged info about the failed job.

Reply:
Not sure what that would look like.

Reply:
@jofsky pytest has a `caplog` fixture which can be used to capture logs and make assertions about them: https://docs.pytest.org/en/stable/logging.html#caplog-fixture

Here is an example: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/tasks/bulkdata/tests/test_mapping_parser.py#L42